### ISSUE TITLE
[Linux only] sys_net::setsockopt aborted: Invalid socket option level!

### ISSUE THREAD

**Issue by giovannilaura** [RPCS3/rpcs3#2788]
Here an error wich occurs only on Linux but not on windows the game tested is BOMBERMAN ULTRA 
log is attached
SYstem AMD RYZEN 1700x RAdeon RX480 Mesa 17.05 Fedorac
[RPCS3BOMBERMAN.txt](https://github.com/RPCS3/rpcs3/files/1002137/RPCS3BOMBERMAN.txt)
core 25 x86_64
Cheers

**Comment by hcorion** [comment]
I was able to reproduce the issue on Arch Linux 64-Bit with Nvidia GTX 1070 and Inte i7 4770

**Comment by mrcmunir** [comment]
Yep I can confirm this by default but if select manually 
libnet.sprx
libnetctl.sprx
libsysutil.sprx

Returned sys_mutex_lock failed 0x80010005 CELL_ESRCH

**Comment by AniLeo** [comment]
This is an issue with HLE, and that mutex error is probably unrelated. Always attach a full rpcs3.log file

**Comment by hcorion** [comment]
Alright, I ran bomberman under Windows, and I got
```
·W 0:00:19.482001 {PPU[0x1000000] Thread (main_thread) [0x002f63e0]} libnet: socket(family=2, type=6, protocol=0)
·W 0:00:19.482010 {PPU[0x1000000] Thread (main_thread) [0x002f63e0]} libnet: SOCK_DGRAM_P2P is not properly implemented.
·E 0:00:19.482019 {PPU[0x1000000] Thread (main_thread) [0x002f63e0]} libnet: Unknown/illegal socket error: 10093
(in file C:\rpcs3\rpcs3\Emu\Cell\Modules\sys_net.cpp:207)
·E 0:00:19.482026 {PPU[0x1000000] Thread (main_thread) [0x002f63e0]} libnet: socket(): error 2
·W 0:00:19.482034 {PPU[0x1000000] Thread (main_thread) [0x002f63f8]} libnet: socketselect(nfds=0, readfds=*0xd0020830, writefds=*0x0, exceptfds=*0x0, timeout=*0xd0020820)
```
And for reference, here is the output on linux:
```
·W 0:00:13.006355 {PPU[0x1000000] Thread (main_thread) [0x001ab3e0]} libnet: socket(family=2, type=6, protocol=0)
·W 0:00:13.006359 {PPU[0x100000d] Thread (EnumAchievement) [0x001aaa68]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=6, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:13.006359 {PPU[0x1000000] Thread (main_thread) [0x001ab3e0]} libnet: SOCK_DGRAM_P2P is not properly implemented.
·W 0:00:13.006374 {PPU[0x100000d] Thread (EnumAchievement) [0x001aaa68]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=7, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:13.006388 {PPU[0x100000d] Thread (EnumAchievement) [0x001aaa68]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=8, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:13.006406 {PPU[0x100000d] Thread (EnumAchievement) [0x001aaa68]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=9, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:13.006421 {PPU[0x100000d] Thread (EnumAchievement) [0x001aaa68]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=10, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:13.006430 {PPU[0x1000000] Thread (main_thread) [0x001ab3d0]} libnet: setsockopt(s=0, level=65535, optname=4096, optval=*0xd00209e0, optlen=4)
·W 0:00:13.017734 {PPU[0x1000000] Thread (main_thread) [0x001ab3d0]} PPU: 'sys_net::setsockopt' aborted
·! 0:00:13.017819 {PPU[0x1000000] Thread (main_thread) [0x001ab3d0]} 
```

**Comment by hcorion** [comment]
Hmm, I've done a lot of digging, and I've came to the conclusion that Linux may actually be right, and Windows isn't setup right, because it's hitting error 10093 (I think) which is 
https://msdn.microsoft.com/en-us/library/windows/desktop/ms740668(v=vs.85).aspx#WSANOTINITIALISED

**Comment by clienthax** [comment]
Interesting, wonder if theres any samples that hit this.

**Comment by clienthax** [comment]
with the above patch on windows
```
·W 0:00:11.324458 {PPU[0x1000000] Thread (main_thread) [0x002f4d60]} sys_ppu_thread: _sys_ppu_thread_create(thread_id=*0x201fefa8, param=*0xd00206e0, arg=0x0, unk=0x0, prio=256, stacksize=0x4000, flags=0x0, threadname=“EnumAchievement”)
·W 0:00:11.324531 {PPU[0x1000000] Thread (main_thread) [0x001aa270]} sceNp: sceNpManagerGetStatus(status=*0xd00208c0)
·W 0:00:11.324632 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=0, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.324733 {PPU[0x1000000] Thread (main_thread) [0x001aaea0]} libnet: socket(family=2, type=6, protocol=0)
·W 0:00:11.324739 {PPU[0x1000000] Thread (main_thread) [0x001aaea0]} libnet: SOCK_DGRAM_P2P is not properly implemented.
·W 0:00:11.324766 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=1, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.324837 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=2, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.324902 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=3, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.324960 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=4, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325021 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=5, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325078 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=6, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325126 {PPU[0x1000000] Thread (main_thread) [0x001aae90]} libnet: setsockopt(s=0, level=65535, optname=4096, optval=*0xd0020810, optlen=4)
·W 0:00:11.325132 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=7, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325135 {PPU[0x1000000] Thread (main_thread) [0x001aae90]} libnet: Socket option OP_SO_USECRYPTO is unimplemented
·E 0:00:11.325150 {PPU[0x1000000] Thread (main_thread) [0x001aae90]} libnet: Unknown/illegal socket error: 0
(in file Emu\Cell\Modules\sys_net.cpp:207)
·E 0:00:11.325158 {PPU[0x1000000] Thread (main_thread) [0x001aae90]} libnet: setsockopt(): error 276756984
·W 0:00:11.325166 {PPU[0x1000000] Thread (main_thread) [0x001aaea8]} libnet: socketclose(s=0)
·W 0:00:11.325190 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=8, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325210 {PPU[0x1000000] Thread (main_thread) [0x001aaeb8]} libnet: socketselect(nfds=0, readfds=*0xd0020640, writefds=*0x0, exceptfds=*0x0, timeout=*0xd0020630)
·W 0:00:11.325246 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=9, details=*0xd007c8f0, data=*0xd007c8e0)
·W 0:00:11.325295 {PPU[0x1000000] Thread (main_thread) [0x002f4f74]} sys_lwmutex: _sys_lwmutex_create(lwmutex_id=*0xd00208d0, protocol=0x1, control=*0x28894f90, arg4=0x80000001, name=0x426f6d6265724d78, arg6=0x0)
·W 0:00:11.325301 {PPU[0x100000d] Thread (EnumAchievement) [0x001aabf0]} sceNpTrophy: sceNpTrophyGetTrophyInfo(context=0x1, handle=0x1, trophyId=10, details=*0xd007c8f0, data=*0xd007c8e0)
·! 0:00:11.325433 {PPU[0x100000d] Thread (EnumAchievement)} Thread time: 0.000000s (0.003449Gc); Faults: 0 [rsx:0, spu:0];
```

**Comment by AniLeo** [comment]
Solved by #3098
